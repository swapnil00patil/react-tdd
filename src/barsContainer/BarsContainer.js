import React from 'react'
import styled from 'styled-components'

import { getBars } from './bars.service'
import BarControls from './components/BarControls'
import Dropdown from './components/Dropdown'
import ProgressBars from './components/ProgressBars'

const Wrap = styled.div`
  width:500px;
  background: #fff;
  padding:18px;
`
const Title = styled.h2`
  text-align:center;
`

class BarsContainer extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      buttons: [],
      bars: [],
      selectedBar: 0
    }
  }

  async componentDidMount () {
    try {
      let response = await getBars()
      if (response) {
        this.setState({
          buttons: response.data.buttons,
          bars: response.data.bars
        })
      }
    } catch (error) {
      throw new Error(error)
    }
  }

  updateProgress (value) {
    console.log(value)
  }

  selectBar (e) {
    console.log(e.target.value)
  }

  render () {
    return (
      <Wrap>
        <Title>Progress Bars Demo</Title>
        <ProgressBars bars={this.state.bars} />
        <BarControls buttons={this.state.buttons} updateProgress={this.updateProgress} />
        <Dropdown bars={this.state.bars} selectBar={this.selectBar} />
      </Wrap>
    )
  }
}

export default BarsContainer
