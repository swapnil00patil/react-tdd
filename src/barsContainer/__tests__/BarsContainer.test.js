import React from 'react'
import { shallow } from 'enzyme'

import BarsContainer from '../BarsContainer'

describe('BarsContainer', () => {
  it('BarsContainer should be defined', () => {
    expect(BarsContainer).toBeDefined()
  })

  it('should contain correct title', () => {
    expect(shallow(<BarsContainer />).contains('Progress Bars Demo')).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<BarsContainer />)
    expect(wrapper).toMatchSnapshot()
  })
})
