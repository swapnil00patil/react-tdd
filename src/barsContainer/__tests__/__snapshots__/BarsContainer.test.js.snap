// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarsContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarsContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.h2>
          Progress Bars Demo
</styled.h2>,
        <ProgressBars
          bars={Array []}
/>,
        <BarControls
          buttons={Array []}
          updateProgress={[Function]}
/>,
        <Dropdown
          bars={Array []}
          selectBar={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Progress Bars Demo",
        },
        "ref": null,
        "rendered": "Progress Bars Demo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bars": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttons": Array [],
          "updateProgress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bars": Array [],
          "selectBar": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h2>
            Progress Bars Demo
</styled.h2>,
          <ProgressBars
            bars={Array []}
/>,
          <BarControls
            buttons={Array []}
            updateProgress={[Function]}
/>,
          <Dropdown
            bars={Array []}
            selectBar={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Progress Bars Demo",
          },
          "ref": null,
          "rendered": "Progress Bars Demo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bars": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttons": Array [],
            "updateProgress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bars": Array [],
            "selectBar": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
